tests/unit/test_cursorManager.py

```diff
@@ -66,6 +66,81 @@ def test_selectForwardThenSelBackward(self):
 		cm.script_selectWord_back(None) # "b" unselected, "a" selected
 		self.assertEqual(cm.selectionOffsets, (0, 1)) # "a" selected
 
+	def test_selectBackwardThenSelForward(self):
+		"""Test selecting backward, then selecting forward without unselecting.
+		Same as test_selectForwardThenSelBackward, but with reversed directions.
+		"""
+		cm = CursorManager(text="abc", selection=(2, 2)) # Caret at "c"
+		cm.script_selectCharacter_back(None) # "b" selected
+		cm.script_selectWord_forward(None) # "b" unselected, "c" selected
+		self.assertEqual(cm.selectionOffsets, (2, 3)) # "c" selected
+
+	def test_selectForwardThenSelBackwardThenUnsel(self):
+		"""Test selecting forward, then selecting backward without unselecting, then unselecting forward.
+		Depends on behavior tested by test_selectForwardThenSelBackward.
+		"""
+		cm = CursorManager(text="abc", selection=(1, 1)) # Caret at "b"
+		cm.script_selectCharacter_forward(None) # "b" selected
+		cm.script_selectWord_back(None) # "b" unselected, "a" selected
+		cm.script_selectCharacter_forward(None) # "a" unselected
+		self.assertEqual(cm.selectionOffsets, (1, 1)) # Caret at "b", no selection
+
+	def test_selectBackwardThenSelForwardThenUnsel(self):
+		"""Test selecting backward, then selecting forward without unselecting, then unselecting backward.
+		Same as test_selectForwardThenSelBackwardThenUnsel, but with reversed directions.
+		Depends on behavior tested by test_selectBackwardThenSelForward.
+		"""
+		cm = CursorManager(text="abc", selection=(2, 2)) # Caret at "c"
+		cm.script_selectCharacter_back(None) # "b" selected
+		cm.script_selectWord_forward(None) # "b" unselected, "c" selected
+		cm.script_selectCharacter_back(None) # "c" unselected
+		self.assertEqual(cm.selectionOffsets, (2, 2)) # Caret at "c", no selection
+

```

Comment:
> There is no coverage for  selectBackwardThenUnselThenSelForward  but there is for selectForwardThenUnselThenSelBackward ?

This was an oversight when I wrote the original batch of tests. I shall fix.

> Can these tests be broken down? Is there state that can only be achieved by using the script methods?

Yes; there is state here that is only set by the methods. I could set it manually for testing; it's just a boolean, though it's not something that should normally be tweaked by a caller. However, a big part of what we're testing here is that the state gets set correctly throughout these various movements. I'm concerned we might miss something if we initialise state directly, especially since it's already somewhat hard to grasp how one might get into various states.